tests/services/test_url_security.py

```
@@ -0,0 +1,131 @@
+    def test_noop_for_empty(self):
+        validate_no_ssrf(None, allow_private=False)
+        validate_no_ssrf("", allow_private=False)
+
+
+class TestSchemaValidation:
+    """Default config allows private (ALLOW_PRIVATE_INTEGRATION_URLS=True)."""
+
+    def test_paperless_accepts_private_by_default(self):
+        data = PaperlessConnectionData(
+            paperless_url="http://127.0.0.1:8000",
+            paperless_api_token="dummytoken123",
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to argument: "paperless_api_token"

(S106)